src/main/java/sirius/web/services/ServiceDispatcher.java

```diff
@@ -108,6 +103,16 @@ private boolean doDispatch(WebContext ctx) {
     private void invokeService(WebContext ctx, ServiceCall call, StructuredService serv) {
         TaskContext.get().setSystem(SYSTEM_SERVICE).setSubSystem(serv.getClass().getSimpleName());
 
+        // Check firewall
+        if (firewall != null && !serv.getClass().isAnnotationPresent(Unlimited.class)) {
+            if (firewall.handleRateLimiting(ctx,
+                                            Optional.ofNullable(serv.getClass().getAnnotation(Limited.class))
+                                                    .map(Limited::value)
+                                                    .orElse(Limited.HTTP))) {
+                return;
```

Comment:
HTTP Status Code 429 "Too many requests" statt einfach ins leere laufen zu lassen?

Reply:
If handle returns true, we expect the firewall to have the request handled (e.g. with said status code)